src/TypeInference/Env.mli

```diff
@@ -92,6 +92,9 @@ val lookup_var : t -> S.var S.path -> Module.var_info option
 val lookup_implicit :
   t -> S.var S.path -> (T.var * T.scheme * (Position.t -> unit)) option
 
+(* TODO: Add description *)
```

Comment:
As mentioned in dc we're unable to write it, could you provide it, please?

Reply:
Now, I understand. This is private function of `Env` module that is used to transform scheme of `label` to actual label type. I assume that it became part of the public interface, because it is used in `open_scheme`. But I think the `scheme_to_label` function is too ugly to be public. I suggest to remove it from the interface, but add another function that extends the environment with the label of given scheme (it could be named `Env.add_the_label'`, `Env.add_the_label_sch`, or something like that). It should be implemented as the following.
```
let add_the_label_sch env sch =
  let (eff, tp0, eff0) = scheme_to_label sch in
  add_the_label env eff tp0 eff0
```
Then, the `open_scheme` function could be changed to used `Env.add_the_label_sch`, which would be much cleaner.

Reply:
Done

Reply:
The documentation comment of `Env.add_the_label_sch` could be better. First of all, it's "the label" instead of "a label": there is at most one distinguished label on which handlers handle. Moreover this label is not present in given scheme, but it has given scheme. This scheme must be monomorphic label type (`TLabel`).

Reply:
Changed it to:` (** Extend an environment with the label of a given scheme. *)`
Is that alright?

Reply:
Also, shouldn't it be the environment? Since we are passing a concrete one to every function.

Reply:
Now, the comment is right. When I said that there is at most one, I mean that each "the label" shadows the previous one.